src/stan/analyze/mcmc/compute_potential_scale_reduction.hpp

```diff
@@ -124,6 +275,40 @@ inline double compute_potential_scale_reduction(
   return compute_potential_scale_reduction(draws, sizes);
 }
 
+/**
+ * Computes the potential scale reduction (Rhat) using rank based diagnostic for
+ * the specified parameter across all kept samples. Based on paper
+ * https://arxiv.org/abs/1903.08008
+ *
+ * When the number of total draws N is odd, the (N+1)/2th draw is ignored.
+ *
+ * See more details in Stan reference manual section "Potential
+ * Scale Reduction". http://mc-stan.org/users/documentation
+ *
+ * Current implementation assumes draws are stored in contiguous
+ * blocks of memory.  Chains are trimmed from the back to match the
+ * length of the shortest chain.
+ *
+ * @param draws stores pointers to arrays of chains
+ * @param sizes stores sizes of chains
+ * @return potential scale reduction for the specified parameter
+ */
+inline double compute_split_potential_scale_reduction_rank(
+    std::vector<const double*> draws, std::vector<size_t> sizes) {
```

Comment:
We want these arguments to come in as constant references. As written this will make a hard copy of the input vectors when you call this function. Making the arguments references (`&`) means the function will just use the already existing object without making a copy and `const` means the arguments will be constant in the function (i.e. we will not modify them)

```suggestion
inline double compute_split_potential_scale_reduction_rank(
    const std::vector<const double*>& draws, const std::vector<size_t>& sizes) {
```

We want containers (like `std::vector` or `Eigen::MatrixXd` or `std::map`) to be passed by constant reference. Small types such as `double`, `int`, `std::size_t` etc. can be passed by value as making copies of them is trivial. Happy to explain this more if you like but don't want to overload you with info. A nice place to read about things like this is Scott Meyers "[Effective Modern C++](https://www.amazon.com/Effective-Modern-Specific-Ways-Improve/dp/1491903996)" which if you google should be easy to find a free copy of online. 

This comment applies to all the function signatures you added here